test/test.js

```diff
@@ -213,11 +219,51 @@ describe('Unhandled exceptions', function () {
       throw new TypeError(message);
     } catch(e) {
       window.onerror(message, 'test.js', 42, 42, e);
-      expect(originalOnErrorCalled).to.be.true;
-      done();
+
+      setTimeout(function(){
+        expect(originalOnErrorCalled).to.be.true;
+        done();
+      }, 10);
```

Comment:
What is the reason for introducing this timeout here?

Reply:
I believe it's needed on each of these due to the: `StackTrace.fromError(err)` being promise-based. The first new promise rejection test assertion was picking up on the result of the final uncaught exception test causing it to fail. This timeout, like the others, allows the thread to release and let that promise "resolve" during the test where it's supposed to resolve.